parrot/src/pfs_dispatch.cc

```diff
@@ -113,9 +113,11 @@ extern int parrot_dir_fd;
 extern int *pfs_syscall_totals32;
 
 int pfs_dispatch_prepexe (struct pfs_process *p, char exe[PATH_MAX], const char *physical_name);
-int pfs_dispatch_isexe( const char *path );
+int pfs_dispatch_isexe( const char *path, uid_t *uid, gid_t *gid );
 
 #define POINTER( i ) ((void *)(uintptr_t)(i))
+#define PRIVILEGED_ID(r,e,s) (((r) == 0) || ((e) == 0) || ((s) == 0))
+#define ALLOWED_ID(n,r,e,s) ((n) == (uid_t) -1 || ((n) == (r)) || ((n) == (e)) || ((n) == (s)))
```

Comment:
I think these macros could be simplified to:

`#define PRIVILEGED_ID(p) (((p->ruid) == 0) || ((p->euid) == 0) || ((p->suid) == 0))`

ditto for `ALLOWED_ID`. Let's also move the two macros to `pfs_dispatch.h`.